code/modules/antagonists/wizard/equipment/spellbook_entries/perks.dm

```diff
@@ -0,0 +1,239 @@
+		if(!istype(generate_perk))
+			continue
+		if(istype(generate_perk, src))
+			continue
+		if(check_perks && is_type_in_list(generate_perk, check_perks.perks))
+			continue
+		if(!generate_perk.buy_spell(user, book, log_buy))
+			continue
+		only_two++
+		if(only_two >= 2)
+			break
+	if(only_two < 2)
+		to_chat(user, span_warning("Gamble cannot give 2 perks, so points are returned"))
+		return FALSE
+
+/datum/spellbook_entry/perks/heart_eater
+	name = "Heart Eater"
+	desc = "Gives you ability to obtain a person's life force by eating their heart. \
+	By eating someone's heart you can increase your maximum health or decrease it but get a random mutation."
+
+/datum/spellbook_entry/perks/heart_eater/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	user.AddComponent(/datum/component/heart_eater)
+
+/datum/spellbook_entry/perks/slime_friends
+	name = "Slime Friends"
+	desc = "Slimes are your friends. \
+	Every 15 seconds you lose some nutriments and summon a random evil slime to fight on your side."
+
+/datum/spellbook_entry/perks/slime_friends/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	user.AddComponent(/datum/component/slime_friends)
+
+/datum/spellbook_entry/perks/transparence
+	name = "Transparence"
+	desc = "You become a little closer to the world of the dead. \
+	Projectiles pass through you, but you lose 25% of your health and you are hunted by an evil stalker who wants to take you to the world of the dead"
+
+/datum/spellbook_entry/perks/transparence/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	if(user.alpha > 125)
+		user.alpha = 125
+	user.maxHealth *= 0.75
+	try_to_make_stalker(user)
+	ADD_TRAIT(user, TRAIT_UNHITTABLE_BY_PROJECTILES, REF(src))
+
+/datum/spellbook_entry/perks/transparence/proc/try_to_make_stalker(mob/living/carbon/human/user)
+	if(get_area(user) == GLOB.areas_by_type[/area/centcom/wizard_station])
+		addtimer(CALLBACK(src, PROC_REF(try_to_make_stalker), user), 10 SECONDS)
+		return
+	user.gain_trauma(/datum/brain_trauma/magic/stalker)
+
+/datum/spellbook_entry/perks/magnetism
+	name = "Magnetism"
+	desc = "You get a small gravity anomaly that orbit around you. \
+	Every second attracts unscrewed objects and unscrewed people nearby closer to you."
+
+/datum/spellbook_entry/perks/magnetism/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	var/atom/movable/magnitizm = new /obj/effect/magnitizm(get_turf(user))
+	magnitizm.orbit(user, 20)
+
+/obj/effect/magnitizm
+	name = "magnitizm anomaly"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shield2"
+	var/mob/living/owner
+	var/timer
+
+/obj/effect/magnitizm/New(loc, ...)
+	. = ..()
+	transform *= 0.4
+
+/obj/effect/magnitizm/orbit(atom/A, radius, clockwise, rotation_speed, rotation_segments, pre_rotation)
+	. = ..()
+	if(!isliving(A))
+		return
+	owner = A
```

Comment:
Can't hold a hard reference to a mob, either make it a WEAKREF or use signals to handle nulling it when it is deleted